client/cli/src/arg_enums.rs

```diff
@@ -163,6 +165,40 @@ impl Into<sc_service::config::RpcMethods> for RpcMethods {
 	}
 }
 
+impl RpcMethods {
+	/// Choose an RPC interface based on the RPC method chosen by the user and a few flags:
+	/// `is_external`, `is_unsafe_external` and `is_validator`.
+	pub fn choose_rpc_interface(
+		self,
+		is_external: bool,
```

Comment:
Taking 3 bools is really error prone.

Maybe some enums?

Reply:
They come directly from flags so an enum would make the user write more code and will be pretty annoying. A struct is the simplest and more okay solution. Or a builder pattern. But in my opinion this is all overkill for CLI stuff.

Reply:
could also use bit flags :D